pkg/gofr/container/mockcontainer_test.go

```diff
@@ -54,30 +56,103 @@ func Test_HttpServiceMock(t *testing.T) {
+			Name string
+		}
+
+		expectedUser := User{ID: 1, Name: "John"}
+		passedUser := User{}
+		actualUser := User{}
+
+		mock.SQL.ExpectSelect(context.Background(), &passedUser, "SELECT * FROM users WHERE id = ?", 1).ReturnsResponse(expectedUser)
+
+		mockContainer.SQL.Select(context.Background(), &actualUser, "SELECT * FROM users WHERE id = ?", 1)
+		require.Equal(t, expectedUser, actualUser)
+	})
+
+	t.Run("Test with map", func(t *testing.T) {
+		expectedSettings := map[string]int{"a": 1, "b": 2}
+		passedSettings := map[string]int{}
+		actualSettings := map[string]int{}
+
+		mock.SQL.ExpectSelect(context.Background(), &passedSettings, "SELECT * FROM settings").ReturnsResponse(expectedSettings)
+
+		mockContainer.SQL.Select(context.Background(), &actualSettings, "SELECT * FROM settings")
+		require.Equal(t, expectedSettings, actualSettings)
+	})
+}
+
+func TestExpectSelect_ErrorCases(t *testing.T) {
+	mockDB, sqlMock, _ := sql.NewSQLMocks(t)
+	ctrl := gomock.NewController(t)
+	expectation := expectedQuery{}
+	mockLogger := NewMockLogger(ctrl)
+	sqlMockWrapper := &mockSQL{sqlMock, &expectation}
+	sqlDB := &sqlMockDB{mockDB, &expectation, mockLogger}
+	sqlDB.finish(t)
+
+	t.Run("NonPointer_Value_In_ExpectSelect", func(t *testing.T) {
+		mockLogger.EXPECT().Errorf(gomock.Any(), gomock.Any())
+
+		var uninitializedVal, resultVal int
+
+		expectedVal := 123
 
-	mockContainer.SQL.Select(context.Background(), &ids, "select quantity from items where id =?", 123)
-	mockContainer.SQL.Select(context.Background(), &ids, "select quantity from items where id =?", 132)
+		sqlMockWrapper.ExpectSelect(context.Background(), uninitializedVal, "SELECT * FROM test WHERE id=?", 1).ReturnsResponse(expectedVal)
+
+		sqlDB.Select(context.Background(), &resultVal, "SELECT * FROM test WHERE id=?", 1)
+		assert.Zero(t, resultVal)
+	})
+
+	t.Run("PointerValue_In_ReturnsResponse", func(t *testing.T) {
+		mockLogger.EXPECT().Errorf("received different expectations: %v", gomock.Any())
+
+		var uninitializedVal, resultVal int
+
+		expectedVal := 123
+
+		sqlMockWrapper.ExpectSelect(context.Background(), &uninitializedVal, "SELECT * FROM test WHERE id=?", 1).ReturnsResponse(&expectedVal)
+
+		sqlDB.Select(context.Background(), &resultVal, "SELECT * FROM test WHERE id=?", 1)
+		assert.Zero(t, resultVal)
+	})
+
+	t.Run("Type_Mismatch_Between_Expect_And_Response", func(t *testing.T) {
+		mockLogger.EXPECT().Errorf("received different expectations: %v", gomock.Any())
+
+		expectedVal := make([]string, 0)
+		resultVal := make([]string, 0)
+
+		sqlMockWrapper.ExpectSelect(context.Background(), &expectedVal, "SELECT * FROM test WHERE id=?", 1).ReturnsResponse(123)
+
+		sqlDB.Select(context.Background(), &resultVal, "SELECT * FROM test WHERE id=?", 1)
+		assert.Empty(t, resultVal)
+	})
+
+	t.Run("Select_Called_Without_Expectations", func(t *testing.T) {
+		mockLogger.EXPECT().Errorf("did not expect any calls for Select with query: %q", gomock.Any())
+
+		val := make([]string, 0)
```

Comment:
Same as https://github.com/gofr-dev/gofr/pull/1363#discussion_r1903984411


```suggestion
		var val []string 
```